ckanext/cfpb_extrafields/plugin.py

```diff
@@ -35,8 +36,71 @@ def popup_usage_restrictions():
     return "Enter instructions for what users can and cannot do with the data."
 
 class ExampleIDatasetFormPlugin(p.SingletonPlugin, tk.DefaultDatasetForm):
-    p.implements(p.IDatasetForm)
-    p.implements(p.IConfigurer)
+
+    p.implements(p.IResourceController)
+    def before_create(self, context, resource):
+        return
+    def after_create(self, context, resource):
+        # must have changed['format'] when you created the resource
+        tk.redirect_to(controller='package',action='resource_edit',
+                       resource_id=resource['id'],id=resource['package_id'])
+
+    def _control_datadict(self, resource):
+        ''' delete and rebuild datadict'''
+        record = resource.get('datadict','')
+        if record:
+            resource.pop('datadict', None)
+            record = eval(record) 
```

Comment:
be careful with this, it will lead to XSS and injection style attacks.  Let's parse this with a json parser instead of using eval.  That way, if we get something other than json it will fail.  Also, use pop or use get, don't use both.  pop will return the item it removed.  
```
if 'datadict' in resource:
    record = resource.pop('datadict')  # no need for a default, we know it exists!
    record_json = json.loads(record)
    ...
```